include/alpaka/atomic/AtomicCpu.hpp

```diff
@@ -0,0 +1,26 @@
+/* Copyright 2021 Andrea Bocci, Felice Pantaleo
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
```

Comment:
I have not tested it, but this should work:
```suggestion

#ifndef ALPAKA_DISABLE_ATOMIC_ATOMICREF
#    if BOOST_VERSION < 107300
#        warning boost::atomic_ref requires Boost vrsion 1.73 or higher. Please update your version of Boost, or disable the use of boost::atomic_ref with -DALPAKA_DISABLE_ATOMIC_ATOMICREF
#    endif
#    define ALPAKA_DISABLE_ATOMIC_ATOMICREF
#endif

```

You could make CMake set `-DALPAKA_DISABLE_ATOMIC_ATOMICREF` if the version of Boost is not high enough - but I have no idea how to do that :-/

Reply:
Since we now require Boost 1.74 this change is no longer necessary.

Reply:
OK, so nothing is needed on the CMake side.

But there is no check anywhere else in the code that `BOOST_VERSION` is 1.74 or greater, so I think it would be useful to include this check, at least for the people that do not use CMake.